### PR TITLE
Article entity

### PR DESCRIPTION
**Pull request by quantumm**

### CODE REVIEW

File: src/App/Controller/ArticleController.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        $article          = new Article();
+        $numberOfArticles = $proposal->getNumberOfArticles();
+        $form             = $this->createForm(EditArticleType::class, $article);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $article
+                ->setProposal($proposal)
+                ->setNumber($numberOfArticles + 1)
```
Comment (Einenlum) [inline src/App/Controller/ArticleController.php]: Shouldn't it be done automatically in the `setProposal` method?

File: src/App/Controller/ArticleController.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                ->snapshot()
+            ;
+
+            $proposal
+                ->incrementVersionNumber()
```
Comment (Einenlum) [inline src/App/Controller/ArticleController.php]: Same?

File: src/App/Controller/PublicDiscussionController.php [hunk 3]
```diff
@@ -230,6 +267,16 @@ public function deleteAction($slug)
                 'slug' => $proposalSlug,
             ]));
         }
+
+        if ($discussion->isArticleDiscussion()) {
```
Comment (Einenlum) [inline src/App/Controller/PublicDiscussionController.php]: What if the controller does not go into this `if` ?

File: src/App/Controller/PublicDiscussionController.php [hunk 4]
```diff
@@ -339,6 +426,8 @@ public function lockAction($slug)
         $em->persist($discussion);
         $em->flush();
 
+        $this->get('session')->getFlashBag()->add('info', 'The discussion has been locked');
```
Comment (Einenlum) [inline src/App/Controller/PublicDiscussionController.php]: The application is in french but all flashbags are in english?

File: src/App/Controller/ArticleController.php (hunk 2 above)
Comment (quantumm) [reply src/App/Controller/ArticleController.php]: Ok. Everything has been put in the setProposal method.

File: src/App/Controller/PublicDiscussionController.php (hunk 3 above)
Comment (quantumm) [reply src/App/Controller/PublicDiscussionController.php]: An exception has been added after this if.

File: src/App/Controller/PublicDiscussionController.php (hunk 4 above)
Comment (quantumm) [reply src/App/Controller/PublicDiscussionController.php]: Hmm yes for the moment :P We will deal with the translations later

**Pull request merged**